src/ext/java/org/opentripplanner/ext/siri/EntityResolver.java

```diff
@@ -0,0 +1,167 @@
+package org.opentripplanner.ext.siri;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+import org.opentripplanner.transit.model.network.Route;
+import org.opentripplanner.transit.model.organization.Operator;
+import org.opentripplanner.transit.model.site.RegularStop;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.opentripplanner.transit.model.timetable.TripIdAndServiceDate;
+import org.opentripplanner.transit.model.timetable.TripOnServiceDate;
+import org.opentripplanner.transit.service.TransitService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.org.siri.siri20.DatedVehicleJourneyRef;
+import uk.org.siri.siri20.EstimatedVehicleJourney;
+import uk.org.siri.siri20.FramedVehicleJourneyRefStructure;
+import uk.org.siri.siri20.MonitoredVehicleJourneyStructure;
+
+/**
+ * This class is responsible for resolving references to various entities in the transit model for
+ * the SIRI updaters
+ */
+public class EntityResolver {
+
+  private static final Logger LOG = LoggerFactory.getLogger(EntityResolver.class);
+
+  private final TransitService transitService;
+
+  private final String feedId;
+
+  public EntityResolver(TransitService transitService, String feedId) {
+    this.transitService = transitService;
+    this.feedId = feedId;
+  }
+
+  public FeedScopedId resolveId(String entityId) {
+    return new FeedScopedId(feedId, entityId);
+  }
+
+  /**
+   * Resolve a {@link Trip} either by resolving a service journey id from EstimatedVehicleJourney ->
+   * FramedVehicleJourneyRef -> DatedVehicleJourneyRef or a dated service journey id from
+   * EstimatedVehicleJourney -> DatedVehicleJourneyRef.
+   */
+  public Trip resolveTrip(EstimatedVehicleJourney journey) {
+    Trip trip = resolveTrip(journey.getFramedVehicleJourneyRef());
+    if (trip != null) {
+      return trip;
+    }
+
+    if (journey.getDatedVehicleJourneyRef() != null) {
+      String datedServiceJourneyId = journey.getDatedVehicleJourneyRef().getValue();
```

Comment:
This is not DSJ id.

Reply:
It is according to the [nordic profile](https://enturas.atlassian.net/wiki/spaces/PUBLIC/pages/637370392/SIRI-ET#SIRI-ET-EstimatedVehicleJourney)

<img width="1247" alt="image" src="https://user-images.githubusercontent.com/698100/214150101-fe4bc175-32db-4490-9154-9ce2e6219141.png">